src/hooks/VoicePlayer/index.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { createContext, useCallback, useContext, useState } from 'react';
+import {
+  VoicePlayerEvent,
+  VoicePlayerEventParams,
+  VoicePlayerEventStorage,
+  VoicePlayerEventTypes,
+} from './voicePlayerEvent';
+
+const noop = () => {/* noop */ };
+
+// VoicePlayerProvider interface
+export interface VoicePlayerProps {
+  children: React.ReactElement;
+}
+export interface VoicePlayerPlayProps {
+  audioFile: File;
+  playbackTime: number;
+  groupKey: string;
+}
+export interface VoicePlayerContext {
+  play: (props: VoicePlayerPlayProps) => void;
+  stop: (groupKey?: string) => void;
+  addEventHandler: (props: VoicePlayerEvent) => void;
+  removeEventHandler: (groupKey: string, handlerId: string) => void;
+}
+
+const VoicePlayerContext = createContext<VoicePlayerContext>({
+  play: noop,
+  stop: noop,
+  addEventHandler: noop,
+  removeEventHandler: noop,
+});
+
+export const VoicePlayerProvider = ({
+  children,
+}: VoicePlayerProps): React.ReactElement => {
+  const [eventStorage, setEventStorage] = useState<VoicePlayerEventStorage>({});
+  const [currentPlayer, setAudioPlayer] = useState<HTMLAudioElement>(null);
+  const [currentGroupKey, setCurrentGroupKey] = useState<string>('');
+
+  const addEventHandler = (event: VoicePlayerEvent): void => {
```

Comment:
was better to have event_handler as a seperate system, and add tests to it ~
But if you can add tests to voice player, its good enough

Reply:
https://sendbird.atlassian.net/browse/UIKIT-3260